metals/src/main/scala/scala/meta/internal/metals/ServerLivenessMonitor.scala

```diff
@@ -178,17 +135,23 @@ object ServerLivenessMonitor {
   object FirstPing extends State
   object Running extends State
 
-  class ReconnectOptions(optReconnect: Option[() => Future[Unit]]) {
-    val canReconnect = optReconnect.nonEmpty
-    val reconnect: () => Future[Unit] =
-      optReconnect.getOrElse(() => Future.successful(()))
-  }
-
-  object ReconnectOptions {
-    def empty = new ReconnectOptions(None)
-    def apply(reconnect: () => Future[Unit]) = new ReconnectOptions(
-      Some(reconnect)
-    )
-  }
+  def connectedParams(icons: Icons): MetalsStatusParams =
+    MetalsStatusParams(
+      s"BSP ${icons.link}",
+      "info",
+      show = true,
+      tooltip = "Metals is connected to the build server.",
```

Comment:
Perhaps outside the scope of this PR, but would it be possible to include the BSP type (Bloop, SBT, scala-cli, mill, etc.) in the tooltip? :slightly_smiling_face:

Reply:
Sure, that actually not a problem at all.